src/action.ts

```diff
@@ -0,0 +1,122 @@
+import { setOutput as setActionOutput } from '@actions/core';
+import semverIncrement from 'semver/functions/inc';
+import semverDiff from 'semver/functions/diff';
+import type { ReleaseType as SemverReleaseType } from 'semver';
+
+import { getTags } from './git-operations';
+import {
+  FieldNames,
+  getMetadataForAllPackages,
+  getPackagesToUpdate,
+  getPackageManifest,
+  PackageManifest,
+  updatePackage,
+  updatePackages,
+} from './package-operations';
+import { getActionInputs, isMajorSemverDiff, WORKSPACE_ROOT } from './utils';
+
+/**
+ * Action entry function. Gets git tags, reads the work space root package.json,
+ * and updates the package(s) of the repository per the Action inputs.
+ *
+ * @see updateMonorepo - For details on monorepo workflow.
+ * @see updatePolyrepo - For details on polyrepo (i.e. single-package
+ * repository) workflow.
+ */
+export async function main(): Promise<void> {
+  const actionInputs = getActionInputs();
```

Comment:
Nit: This seems like a curious way of dividing responsibilities between here and `index.js`. Both handle action-interface type stuff, with this function being responsible for getting inputs, but `index.js` being responsible for handling errors properly.

Perhaps a better way to divide responsibilities would be to let this module be responsible for updating package(s)? That encompasses everything but this first line here. Then `index.js` can be the "action interface", similar to how a package with a CLI might have a separate `cli.js` module. And this module could have a more descriptive name like `update.ts` or `updatePackages.ts`.